docs/fsharp/style-guide/formatting.md

```diff
@@ -25,6 +25,177 @@ When indentation is required, you must use spaces, not tabs. At least one space
 
+## Formatting blank lines
+
+* Separate top-level function and class definitions with two blank lines.
+* Method definitions inside a class are separated by a single blank line.
+* Extra blank lines may be used (sparingly) to separate groups of related functions. Blank lines may be omitted between a bunch of related one-liners (for example, a set of dummy implementations).
+* Use blank lines in functions, sparingly, to indicate logical sections.
+
+## Formatting comments
+
+Generally prefer multiple double-slash comments over ML-style block comments.
+
+```fsharp
+// Prefer this style of comments when you want
+// to express written ideas on multiple lines.
+
+(*
+    ML-style comments are fine, but not a .NET-ism.
+    They are useful when needing to modify multi-line comments, though.
+*)
+```
+
+Inline comments should capitalize the first letter.
+
+```fsharp
+let f x = x + 1 // Increment by one.
+```
+
+## Naming conventions
+
+### Use camelCase for class-bound, expression-bound and pattern-bound values and functions
+
+It is common and accepted F# style to use camelCase for all names bound as local variables or in pattern matches and function definitions.
+
+```fsharp
+// OK
+let addIAndJ i j = i + j
+
+// Bad
+let addIAndJ I J = I+J
+
+// Bad
+let AddIAndJ i j = i + j
+```
+
+Locally-bound functions in classes should also use camelCase.
+
+```fsharp
+type MyClass() =
+
+    let doSomething () =
+
+    let firstResult = ...
+
+    let secondResult = ...
+
+    member x.Result = doSomething()
+```
+
+### Use camelCase for module-bound public functions
+
+When a module-bound function is part of a public API, it should use camelCase:
+
+```fsharp
+module MyAPI =
+    let publicFunctionOne param1 param2 param2 = ...
+
+    let publicFunctionTwo param1 param2 param3 = ...
+```
+
+### Use camelCase for internal and private module-bound values and functions
+
+Use camelCase for private module-bound values, including the following:
+
+* Ad hoc functions in scripts
+
+* Values making up the internal implementation of a module or type
+
+```fsharp
+let emailMyBossTheLatestResults =
+    ...
+```
+
+### Use camelCase for parameters
+
+All parameters should use camelCase in accordance with .NET naming conventions.
+
+```fsharp
+module MyModule =
+    let myFunction paramOne paramTwo = ...
+
+type MyClass() =
+    member this.MyMethod(paramOne, paramTwo) = ...
+```
+
+### Use PascalCase for modules
+
+All modules (top-level, internal, private, nested) should use PascalCase.
+
+```fsharp
+module MyTopLevelModule
+
+module Helpers =
+    module private SuperHelpers =
+        ...
+
+    ...
+```
+
+### Use PascalCase for type declarations, members, and labels
+
+Classes, interfaces, structs, enumerations, delegates, records, and discriminated unions should all be named with PascalCase. Members within types and labels for records and discriminated unions should also use PascalCase.
+
+```fsharp
+type IMyInterface =
```

Comment:
Do you want to recommend that interfaces start with `I`?

Reply:
Yep. We follow .NET naming guidelines here.